docs/proposals/25-private-cloud-profiles.md

```diff
@@ -0,0 +1,299 @@
+
+### Current State
+
+Cloud Profiles are non namespaces resources. This means that in a typical gardener installation, only a handful of cloud profiles (i.e. one per cloud provider) exist. They can be consumed by any project's clusters. Consequently, when a project requires changes to a cloud profiles for any of the reasons mentioned in the [context section](#context), they are changed for the entire gardener landscape. Since some projects might require frequent changes to a cloud profile, this become quite a cumbersome process. Additionally it adds confusion to every projects workflow if say for example a team requires an extension of the expiration date of a kubernetes version since they will be wondering why it was extended.
+
+### Goals
+
+- Reduce load on Gardener operators for maintaining cloud profiles
+- Possibly reduce the time a team has to wait until a change in a cloud profile is reflected in production
+- Make it so that project-scoped information in the cloud profile is only visible to the relevant project
+- Full backward compatibility
+
+### Non-Goals
+
+- Automate approval process for changes in Gardeners cloud profiles
+- Add different, possibly unsupported Kubernetes versions or machine images versions and names
+
+## Proposal
+
+It is proposed to implement a solution that enables project administrators to create custom (private) cloud profiles. These cloud profiles would be consumed by project users of the project they were created in.
+
+### Approach
+
+First of all, a new, namespaced API object is defined the `PrivateCloudProfile`, it's type definition is very similar to the `CloudProfile` object but it omits the `type`, `providerConfig` and `seedSelector` fields. The `type` field is omitted as it is inherited from the parent cloud profile. The `providerConfig` field is omitted as gardener itself does not know anything about the structure of this field and cannot merge it accordingly. The `seedSelector` field is omitted as it is to complex for this GEP but may be added in the future.
+
+The general approach is that a private cloud profile inherits from a cloud profile using a `parent` field in the private cloud profile and should therefore not change the type or provider config. Fields such as `machineTypes`, `volumeTypes`, `regions` and `caBundle` are going to be merged with the parent cloud profile. However, a restriction need to be defined so that the `kubernetes` and `machineImages` fields in a private cloud profile may only be adjusted by a gardener operator to reduce the chance of a team staying on an unsupported kubernetes version. A similar problem is already solved in gardener using custom rbac verbs [here](https://github.com/gardener/gardener/blob/master/plugin/pkg/global/customverbauthorizer), the approach is described in more detail in the [custom rbac verb section](#custom-rbac-verb).
```

Comment:
Ha, nice that you want to support even this use case (extend the lifetime for a version for a project, but not for everybody - under control of the Gardener operators), but if that's too much effort, it can be discussed. The idea is great, though. The use case definitely exists.

Reply:
Thanks, glad to hear that the use case is valid.